src/main.js

```diff
@@ -1,22 +1,264 @@
 /// <reference types="jquery" />
 
-//1. Construir un pokedex
-// a.buscar por id
-// b.buscar por tipo
-
-
-
-
-
-
-
+document.querySelector('#search-pokemon-button').addEventListener('click', function () {
+    const pokemon = $('#pokemon-id-input').val().toLowerCase();
+    getPokemonHandler(pokemon)
 
+})
+document.querySelector('#previous-page').addEventListener('click', function () {
+    const PREVIOUS_PAGE_OFFSET = 16;
+    const previousPage = firstPokemonOnListId - PREVIOUS_PAGE_OFFSET;
+    getPokemonList(previousPage)
+})
+document.querySelector('#next-page').addEventListener('click', function () {
+    const nextPage = lastPokemonOnListId;
+    getPokemonList(nextPage)
+})
+$('#pokemon-list').on('click', event => {
+    const pokemon = event.target.id;
+    if(!event.target.classList.contains("btn-link")) return;
+    getPokemonHandler(pokemon)
+})
+getPokemonList();
 
+/**
+ * Fetches a list of Pokemon with the given offset and limit.
+ * @param {number} [offset=0] - The offset of the first Pokemon to fetch.
+ * @param {number} [amount=10] - The maximum number of Pokemon to fetch.
+ */
+function getPokemonList(offset = 0, amount = 15){
+    const LOWEST_POKEMON_OFFSET = 0;
+    offset = Math.max(offset, LOWEST_POKEMON_OFFSET);
+    fetch(`https://pokeapi.co/api/v2/pokemon/?offset=${offset}&limit=${amount}`)
+    .then(response => response.json())
+    .then(data => {
+        if(!totalPokemon) { updateTotalPokemon(data.count); };
+        updateFirstAndLastPokemon(offset, amount);
+        updatePokemonList(data.results, offset);
+    })
+    .catch(() => {
+        alert('Could not get pokemons list');
+    })
+}
+function updateTotalPokemon(total){
+    totalPokemon = total;
+    try{
+        Cypress.env('totalPokemon', total);
 
+    } catch(error){
+        console.log(error);
+    }
+}
+function getPokemonHandler(pokemon){
+    clearPokemon();
+    addLoadingToPokemon();
+    hidePokemonNotFoundError()
+    getPokemonByIdOrName(pokemon);
+}
+function addLoadingToPokemon(){
+    $('#pokemon-carousel .carousel-inner')
+    .append(`<div class="loader pokemon-carousel active"></div>`)
+}
+function removeLoadingFromPokemon(){
+    $('#pokemon-carousel .loader.pokemon-carousel').remove();
+}
+let totalPokemon // total amount of pokemons is global por testing purposes
+let firstPokemonOnListId;
+let lastPokemonOnListId;
+function updateFirstAndLastPokemon(offset, amount){
+    firstPokemonOnListId = offset + 1;//first offset is 0 but first pokemon is 1
+    lastPokemonOnListId = offset + amount;
+    if(lastPokemonOnListId > totalPokemon)//makes sure it doesen't go further from last pokemon
+    lastPokemonOnListId = totalPokemon;
+}
```

Comment:
Is it necessary to update the first and last pokemon? can't you evaluate if the offset is 0 the pokemon id is +1, and if it isn't the pokemon id is something like offset*amount +1? instead of doing firstPokemonListId, lastPokemonListId? let's look for another approach, rename the function and refactor what you need.